circuits/benches/valid_wallet_create.rs

```diff
@@ -0,0 +1,86 @@
+//! Tests the process of proving and verifying a `VALID WALLET CREATE` circuit
+#![allow(incomplete_features)]
+#![feature(generic_const_exprs)]
+
+use circuit_types::traits::{CircuitBaseType, SingleProverCircuit};
+use circuits::zk_circuits::valid_wallet_create::ValidWalletCreate;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use merlin::HashChainTranscript;
+use mpc_bulletproof::r1cs::Prover;
+use mpc_bulletproof::PedersenGens;
+use rand::thread_rng;
+
+use circuits::zk_circuits::valid_wallet_create::test_helpers::create_default_witness_statement;
+
+/// Tests the time taken to apply the constraints of `VALID WALLET CREATE` circuit
+pub fn bench_apply_constraints(c: &mut Criterion) {
+    // Build a witness and statement
+    let (witness, statement) = create_default_witness_statement();
+
+    let mut group = c.benchmark_group("constraint-generation");
+    group.bench_function(BenchmarkId::from_parameter(""), |b| {
+        b.iter(|| {
+            let mut rng = thread_rng();
+            let mut transcript = HashChainTranscript::new(b"test");
+            let pc_gens = PedersenGens::default();
+            let mut prover = Prover::new(&pc_gens, &mut transcript);
+
+            let (witness_var, _) = witness.commit_witness(&mut rng, &mut prover);
+            let statement_var = statement.commit_public(&mut prover);
```

Comment:
Should this setup code be part of the benchmarks?

Reply:
moved it to a setup function in `iter_batched`